packages/language-core/lib/parsers/scriptSetupRanges.ts

```diff
@@ -254,6 +257,10 @@ export function parseScriptSetupRanges(
 				};
 
 				if (ts.isVariableDeclaration(parent)) {
+					if (ts.isObjectBindingPattern(parent.name)) {
```

Comment:
Perhaps we'd better move the logic into https://github.com/KazariEX/vuejs-language-tools/blob/8379263ecc99ef550cc61f9567d70467633a8131/packages/language-core/lib/parsers/scriptSetupRanges.ts#L236 and use https://github.com/vuejs/language-tools/blob/dc6f943c6ccb26778906a49532966aaf69fdc5a5/packages/language-core/lib/codegen/common.ts#L44-L68

Reply:
Do you mean to move the all logic into the main function?